base/array.jl

```diff
@@ -247,8 +247,8 @@ function unsafe_copyto!(dest::Array{T}, doffs, src::Array{T}, soffs, n) where T
               pointer(dest, doffs), pointer(src, soffs), n * Base.bitsunionsize(T))
         # copy selector bytes
         ccall(:memmove, Ptr{Cvoid}, (Ptr{Cvoid}, Ptr{Cvoid}, UInt),
-              convert(Ptr{UInt8}, pointer(dest)) + length(dest) * Base.bitsunionsize(T) + doffs - 1,
-              convert(Ptr{UInt8}, pointer(src)) + length(src) * Base.bitsunionsize(T) + soffs - 1,
+              ccall(:jl_array_typetagdata, Ptr{UInt8}, (Any,), dest) + doffs - 1,
+              ccall(:jl_array_typetagdata, Ptr{UInt8}, (Any,), src) + soffs - 1,
```

Comment:
For future readers: I should probably add an `@noinline` to this method, since if our compiler ever decided to inline this method into another function that loads or stores any value from either array, we semantically (at the Julia level, with our TBAA rules) might allow LLVM to replace that function and the contents of the arrays with garbage.

Reply:
Shouldn't that be a commend in the source code so that it can be found by the reader who needs it?

Reply:
Probably, I'll try to add it eventually, although if you decide to try to use `ccall` and/or `unsafe_load` / `unsafe_store` / `pointer_from_objref` / `llvmcall` etc. to bypass Julia semantics, all bets are off anyways typically, so it seems fairly unimportant right now.